### PR TITLE
feat(sql-splitter): support SqlCommentProcessor to split sql by stream

### PR DESCRIPTION
**Pull request by LuckyPickleZZ**
#### What type of PR is this?
type-feature

#### What this PR does / why we need it:
The `SqlCommentProcessor` needs to read SQL text into memory at once and then split it. When the SQL file is large, OOM exceptions may occur. 
This PR supports reading and splitting SQL through a `InputStream`.

#### Which issue(s) this PR fixes:

none, just an optimization

#### Special notes for your reviewer:
The UT has passed.

#### Additional documentation e.g., usage docs, etc.:

```docs

```

### CODE REVIEW

File: server/odc-core/src/main/java/com/oceanbase/odc/core/sql/split/SqlCommentProcessor.java [hunk 1]
```diff
@@ -599,4 +609,71 @@ public int getValue() {
         }
     }
 
+    public class SqlIterator implements Iterator<String>, AutoCloseable {
```
Comment (yhilmare) [inline server/odc-core/src/main/java/com/oceanbase/odc/core/sql/split/SqlCommentProcessor.java:612]: don't do this, make `SqlCommentProcessor` as an util.

the internal status of `SqlCommentProcessor` is much too complex that should not care about it.

from my point of view, `addLineMySQL` and `addLineOracle` is enough

Comment (yhilmare) [inline server/odc-core/src/main/java/com/oceanbase/odc/core/sql/split/SqlCommentProcessor.java:612]: further more, the name of this class may be `SqlLine`? user will get a line from every iteration.

Comment (LuckyPickleZZ) [reply server/odc-core/src/main/java/com/oceanbase/odc/core/sql/split/SqlCommentProcessor.java]: done. Used a static method `getSqlLines` to return inner static class `SqlLine`. The object of `SqlLine` will create a new `SqlCommentProcessor` for usage.

Comment (LuckyPickleZZ) [reply server/odc-core/src/main/java/com/oceanbase/odc/core/sql/split/SqlCommentProcessor.java]: done. Renamed to `SqlLine`

File: server/odc-core/src/main/java/com/oceanbase/odc/core/sql/split/SqlCommentProcessor.java [hunk 2]
```diff
@@ -599,4 +612,77 @@ public int getValue() {
         }
     }
 
+    public static class SqlLine implements Iterator<String>, AutoCloseable {
```
Comment (yizhouxw) [inline server/odc-core/src/main/java/com/oceanbase/odc/core/sql/split/SqlCommentProcessor.java:615]: may `SqlStatementIterator`

**Review by yizhouxw — APPROVED** [review]
lgtm

**Pull request merged**